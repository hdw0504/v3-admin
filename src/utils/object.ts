import { typeIs } from '@/utils/is'

export function deepClone<T>(obj: T): T {
  const copiedObjects = new WeakMap<object, any>()
  const copy = (obj: T): T => {
    const result: any = {}
    switch (typeIs(obj)) {
      case 'Array':
        return (obj as T[]).map(copy) as T
      case 'Date':
        return new Date((obj as Date).getTime()) as T
      case 'Object':
        if (copiedObjects.has(obj as object))
          return copiedObjects.get(obj as object)

        copiedObjects.set(obj as object, result)
        for (const key in obj) {
        // if(obj.hasOwnProperty(key)){
          if (Object.prototype.hasOwnProperty.call(obj, key))
            result[key] = copy(obj[key] as any)
        }
        return result
      default:
        return obj
    }
  }
  return copy(obj)
}

