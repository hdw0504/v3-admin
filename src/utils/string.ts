import type { MaybeRef } from '@vueuse/core'
/**
 * string to lower case
 * @param str strinng
 * @param startWith string
 * @param endWith string
 * @returns string
 */
export function lowerCaseStr(str: string, startWith = '', endWith = '') {
  str = str.replace(/[A-Z]/g, v => `${startWith}${v.toLocaleLowerCase()}${endWith}`)
  if (startWith !== '' && str.startsWith(startWith))
    str = str.substring(1)
  if (endWith !== '' && str.endsWith(endWith))
    str = str.substring(0, str.length - 1)
  return str
}

export function hump2line(str: MaybeRef<string>) {
  return lowerCaseStr(unref(str), '-')
}
