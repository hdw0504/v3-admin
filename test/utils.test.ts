import { describe, it } from 'vitest'
import {
  getAllBreadcrumbList as getAllBreadcrumbListOld,
  getFlatArr as getFlatArrOld,
  getKeepAliveRouterName as getKeepAliveRouterNameOld,
  getMenuListPath as getMenuListPathOld,
  getShowMenuList as getShowMenuListOld,
} from '@/utils/util'
import DynamicRouter from '@/assets/json/dynamicRouter.json'
import { isArray } from '@/utils/is'
import { deepClone } from '@/utils/object'

describe('utils function', () => {
  function getFlatArr(menulist: Menu.MenuOptions[]) {
    return menulist.reduce((prev: Menu.MenuOptions[], current: Menu.MenuOptions) => {
      prev.push(current)
      isArray(current.children) && prev.push(...getFlatArr(current.children))
      return prev
    }, [])
  }

  function getKeepAliveRouterName(menuList: Menu.MenuOptions[]) {
    return menuList.reduce((prev: string[], current: Menu.MenuOptions) => {
      current.meta.isKeepAlive && current.name && prev.push(current.name)
      isArray(current.children) && prev.push(...getKeepAliveRouterName(current.children))
      return prev
    }, [])
  }

  function getShowMenuList(menuList: Menu.MenuOptions[]) {
    const list = deepClone(menuList)
    const loop = (menus: Menu.MenuOptions[]) => {
      return menus.filter((item) => {
        if (isArray(item.children))
          item.children = loop(item.children)
        return !item.meta.isHide
      })
    }
    return loop(list)
  }

  function getMenuListPath(menuList: Menu.MenuOptions[]) {
    return menuList.reduce((prev: string[], cur) => {
      prev.push(cur.path)
      isArray(cur.children) && prev.push(...getMenuListPath(cur.children))
      return prev
    }, [])
  }

  function getAllBreadcrumbList(menuList: Menu.MenuOptions[], result: Record<string, any> = {}, path = []) {
    for (const item of menuList) {
      result[item.path] = [...path, item]
      if (item.children)
        getAllBreadcrumbList(item.children, result, result[item.path])
    }
    return result
  }

  it('getFlatArr', () => {
    expect(getFlatArrOld(DynamicRouter.data)).toEqual(getFlatArr(DynamicRouter.data))
  })

  it('getKeepAliveRouterName', () => {
    expect(getKeepAliveRouterNameOld(DynamicRouter.data)).toEqual(getKeepAliveRouterName(DynamicRouter.data))
  })

  it('getKeepAliveRouterName', () => {
    expect(getShowMenuListOld(DynamicRouter.data)).toEqual(getShowMenuList(DynamicRouter.data))
  })

  it('getMenuListPath', () => {
    expect(getMenuListPathOld(DynamicRouter.data)).toEqual(getMenuListPath(DynamicRouter.data))
  })

  it('getAllBreadcrumbList', () => {
    // expect(getAllBreadcrumbListOld(DynamicRouter.data)).toEqual(getAllBreadcrumbList(DynamicRouter.data))
    expect(getAllBreadcrumbListOld(DynamicRouter.data)).toMatchSnapshot()
    expect(getAllBreadcrumbList(DynamicRouter.data)).toMatchSnapshot()
  })
})
