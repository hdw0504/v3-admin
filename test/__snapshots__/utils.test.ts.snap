// Vitest Snapshot v1

exports[`utils function > getAllBreadcrumbList 1`] = `
{
  "/form": [
    {
      "children": [
        {
          "component": "/form/customFrom/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 form",
          },
          "name": "customFrom",
          "path": "/form/customFrom",
        },
      ],
      "meta": {
        "icon": "Tickets",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "form 表单",
      },
      "name": "form",
      "path": "/form",
      "redirect": "/form/customFrom",
    },
  ],
  "/form/customFrom": [
    {
      "children": [
        {
          "component": "/form/customFrom/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 form",
          },
          "name": "customFrom",
          "path": "/form/customFrom",
        },
      ],
      "meta": {
        "icon": "Tickets",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "form 表单",
      },
      "name": "form",
      "path": "/form",
      "redirect": "/form/customFrom",
    },
    {
      "component": "/form/customFrom/index",
      "meta": {
        "icon": "SetUp",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "配置化 form",
      },
      "name": "customFrom",
      "path": "/form/customFrom",
    },
  ],
  "/home/index": [
    {
      "component": "/home/index",
      "meta": {
        "icon": "HomeFilled",
        "isAffix": true,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "首页",
      },
      "name": "home",
      "path": "/home/index",
    },
  ],
  "/link": [
    {
      "children": [
        {
          "component": "/link/github/index",
          "meta": {
            "icon": "Share",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "https://github.com/hdw0504/v3-admin",
            "title": "GitHub 仓库",
          },
          "name": "github",
          "path": "/link/github",
        },
      ],
      "meta": {
        "icon": "Paperclip",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "外部链接",
      },
      "name": "link",
      "path": "/link",
      "redirect": "/link/github",
    },
  ],
  "/link/github": [
    {
      "children": [
        {
          "component": "/link/github/index",
          "meta": {
            "icon": "Share",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "https://github.com/hdw0504/v3-admin",
            "title": "GitHub 仓库",
          },
          "name": "github",
          "path": "/link/github",
        },
      ],
      "meta": {
        "icon": "Paperclip",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "外部链接",
      },
      "name": "link",
      "path": "/link",
      "redirect": "/link/github",
    },
    {
      "component": "/link/github/index",
      "meta": {
        "icon": "Share",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "https://github.com/hdw0504/v3-admin",
        "title": "GitHub 仓库",
      },
      "name": "github",
      "path": "/link/github",
    },
  ],
  "/other": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
  ],
  "/other/canvas": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/canvas/index",
      "meta": {
        "icon": "Picture",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "canvas",
      },
      "name": "canvas",
      "path": "/other/canvas",
    },
  ],
  "/other/comment": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/comment/index",
      "meta": {
        "icon": "Comment",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "组件通信",
      },
      "name": "Comment",
      "path": "/other/comment",
    },
  ],
  "/other/loop": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/loop/index",
      "meta": {
        "icon": "Box",
        "isAffix": true,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "递归组件",
      },
      "name": "loop",
      "path": "/other/loop",
    },
  ],
  "/other/vueuse": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/vueuse/index",
      "meta": {
        "icon": "HotWater",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "vueuse",
      },
      "name": "vueuse",
      "path": "/other/vueuse",
    },
  ],
  "/table": [
    {
      "children": [
        {
          "component": "/table/default/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 table",
          },
          "name": "tableDefault",
          "path": "/table/default",
        },
      ],
      "meta": {
        "icon": "List",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "table 表格",
      },
      "name": "table",
      "path": "/table",
      "redirect": "/table/default",
    },
  ],
  "/table/default": [
    {
      "children": [
        {
          "component": "/table/default/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 table",
          },
          "name": "tableDefault",
          "path": "/table/default",
        },
      ],
      "meta": {
        "icon": "List",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "table 表格",
      },
      "name": "table",
      "path": "/table",
      "redirect": "/table/default",
    },
    {
      "component": "/table/default/index",
      "meta": {
        "icon": "SetUp",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "配置化 table",
      },
      "name": "tableDefault",
      "path": "/table/default",
    },
  ],
}
`;

exports[`utils function > getAllBreadcrumbList 2`] = `
{
  "/form": [
    {
      "children": [
        {
          "component": "/form/customFrom/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 form",
          },
          "name": "customFrom",
          "path": "/form/customFrom",
        },
      ],
      "meta": {
        "icon": "Tickets",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "form 表单",
      },
      "name": "form",
      "path": "/form",
      "redirect": "/form/customFrom",
    },
  ],
  "/form/customFrom": [
    {
      "children": [
        {
          "component": "/form/customFrom/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 form",
          },
          "name": "customFrom",
          "path": "/form/customFrom",
        },
      ],
      "meta": {
        "icon": "Tickets",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "form 表单",
      },
      "name": "form",
      "path": "/form",
      "redirect": "/form/customFrom",
    },
    {
      "component": "/form/customFrom/index",
      "meta": {
        "icon": "SetUp",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "配置化 form",
      },
      "name": "customFrom",
      "path": "/form/customFrom",
    },
  ],
  "/home/index": [
    {
      "component": "/home/index",
      "meta": {
        "icon": "HomeFilled",
        "isAffix": true,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "首页",
      },
      "name": "home",
      "path": "/home/index",
    },
  ],
  "/link": [
    {
      "children": [
        {
          "component": "/link/github/index",
          "meta": {
            "icon": "Share",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "https://github.com/hdw0504/v3-admin",
            "title": "GitHub 仓库",
          },
          "name": "github",
          "path": "/link/github",
        },
      ],
      "meta": {
        "icon": "Paperclip",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "外部链接",
      },
      "name": "link",
      "path": "/link",
      "redirect": "/link/github",
    },
  ],
  "/link/github": [
    {
      "children": [
        {
          "component": "/link/github/index",
          "meta": {
            "icon": "Share",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "https://github.com/hdw0504/v3-admin",
            "title": "GitHub 仓库",
          },
          "name": "github",
          "path": "/link/github",
        },
      ],
      "meta": {
        "icon": "Paperclip",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "外部链接",
      },
      "name": "link",
      "path": "/link",
      "redirect": "/link/github",
    },
    {
      "component": "/link/github/index",
      "meta": {
        "icon": "Share",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "https://github.com/hdw0504/v3-admin",
        "title": "GitHub 仓库",
      },
      "name": "github",
      "path": "/link/github",
    },
  ],
  "/other": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
  ],
  "/other/canvas": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/canvas/index",
      "meta": {
        "icon": "Picture",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "canvas",
      },
      "name": "canvas",
      "path": "/other/canvas",
    },
  ],
  "/other/comment": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/comment/index",
      "meta": {
        "icon": "Comment",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "组件通信",
      },
      "name": "Comment",
      "path": "/other/comment",
    },
  ],
  "/other/loop": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/loop/index",
      "meta": {
        "icon": "Box",
        "isAffix": true,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "递归组件",
      },
      "name": "loop",
      "path": "/other/loop",
    },
  ],
  "/other/vueuse": [
    {
      "children": [
        {
          "component": "/other/comment/index",
          "meta": {
            "icon": "Comment",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "组件通信",
          },
          "name": "Comment",
          "path": "/other/comment",
        },
        {
          "component": "/other/vueuse/index",
          "meta": {
            "icon": "HotWater",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "vueuse",
          },
          "name": "vueuse",
          "path": "/other/vueuse",
        },
        {
          "component": "/other/canvas/index",
          "meta": {
            "icon": "Picture",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "canvas",
          },
          "name": "canvas",
          "path": "/other/canvas",
        },
        {
          "component": "/other/loop/index",
          "meta": {
            "icon": "Box",
            "isAffix": true,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "递归组件",
          },
          "name": "loop",
          "path": "/other/loop",
        },
      ],
      "meta": {
        "icon": "Document",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "其他",
      },
      "name": "other",
      "path": "/other",
      "redirect": "/other/comment",
    },
    {
      "component": "/other/vueuse/index",
      "meta": {
        "icon": "HotWater",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "vueuse",
      },
      "name": "vueuse",
      "path": "/other/vueuse",
    },
  ],
  "/table": [
    {
      "children": [
        {
          "component": "/table/default/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 table",
          },
          "name": "tableDefault",
          "path": "/table/default",
        },
      ],
      "meta": {
        "icon": "List",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "table 表格",
      },
      "name": "table",
      "path": "/table",
      "redirect": "/table/default",
    },
  ],
  "/table/default": [
    {
      "children": [
        {
          "component": "/table/default/index",
          "meta": {
            "icon": "SetUp",
            "isAffix": false,
            "isFull": false,
            "isHide": false,
            "isKeepAlive": true,
            "isLink": "",
            "title": "配置化 table",
          },
          "name": "tableDefault",
          "path": "/table/default",
        },
      ],
      "meta": {
        "icon": "List",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "table 表格",
      },
      "name": "table",
      "path": "/table",
      "redirect": "/table/default",
    },
    {
      "component": "/table/default/index",
      "meta": {
        "icon": "SetUp",
        "isAffix": false,
        "isFull": false,
        "isHide": false,
        "isKeepAlive": true,
        "isLink": "",
        "title": "配置化 table",
      },
      "name": "tableDefault",
      "path": "/table/default",
    },
  ],
}
`;
